src/core/datatypes/forte_string.cpp

```diff
@@ -69,37 +104,9 @@ int CIEC_STRING::fromString(const char *paValue) {
         continue;
       } else {
         char specialSymbol;
-        switch (inputString[i + 1]) {
-          case '\'':
-            specialSymbol = '\'';
-            break;
-          case 'L':
-          case 'l':
-            specialSymbol = '\x10';
-            break;
-          case 'N':
-          case 'n':
-            specialSymbol = '\n';
-            break;
-          case 'P':
-          case 'p':
-            specialSymbol = '\f';
-            break;
-          case 'R':
-          case 'r':
-            specialSymbol = '\r';
-            break;
-          case 'T':
-          case 't':
-            specialSymbol = '\t';
-            break;
-          case '$':
-            specialSymbol = '$';
-            break;
-          default:
-            DEVLOG_ERROR("Unknown escape symbol %c encountered", inputString[i + 1]);
+        if (getSpecialSymbol(inputString[i + 1], specialSymbol) != -1) {
+          unescapedString += specialSymbol; // append escaped char
```

Comment:
As mentioned before, shouldn't we abort with an error here? @azoitl 
Now we are just not adding the invalid escaped symbol, thereby silently ignoring an error, right?

Reply:
True I missed that. However, giving it a second thought I'm not perfectly sure what should happen here. I had a look in IEC 61131-3 and there nothing is stated about invalid $ escapings. Does anyone know what other languages are doing. Should we just ignore the $ then. I think this is what happens in C or?

Reply:
The compiler gives you an error what I have seen on StackOverflow

Reply:
The previous logic possibly add uninitialized specialSymbol char that trigger compiler warning. I'm also not sure what should happen here.

Reply:
@kumajaya first of all thx for your patience with us. With your work you are definitely improving 4diac FORTE. But as you are poking on a rather important component and as you are changing it, we think it would be good that we get it right this time.

Reply:
@azoitl No problem, sir! I found this great project and started loving it from the beginning because it seemed to suit my needs. Just a little help from my side, I always take compiler warnings seriously and try to understand why they occur but of course I always need good guidance from people who understand more here.

Reply:
@kumajaya it's good you fix compiler warnings, it is important and I fixed a bunch myself, but not always high on our priorities ;)